### PR TITLE
Updated refactoring for the RSpec test suites and the introduction of Alma test fixtures

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #1204 and deprecates #1443

### CODE REVIEW

File: marc_to_solr/lib/princeton_marc.rb [hunk 1]
```diff
@@ -628,132 +628,170 @@ def scsb_doc?(record_id)
 
 # holding block json hash keyed on mfhd id including location, library, call number, shelving title,
 # location note, location has, location has (current), indexes, and supplements
-# pulls from mfhd 852, 866, 867, and 868
-# assumes exactly 1 852 is present per mfhd (it saves the last 852 it finds)
+# pulls from mfhd AVA, 866, 867, and 868
+# assumes exactly 1 AVA is present per mfhd (it saves the last AVA it finds)
```
Comment (christinach) [inline marc_to_solr/lib/princeton_marc.rb:632]: @jrgriffiniii I havent looked the whole Pr but when we index from the mrc files that we export from alma they include the enrichment fields 852. They dont include AVA. The AVA is a tag that comes back only through the Alma API call. Please reverse the changes in princeton_marc file.

File: marc_to_solr/lib/traject_config.rb [hunk 2]
```diff
@@ -1155,57 +1161,101 @@
 end
 
 # Process location code once
-# 852|b and 852|c
+# AVA|b and AVA|c
 each_record do |record, context|
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1165]: Same here. Please reverse the changes. The location_code is a compination of 852|b and 852|c

File: marc_to_solr/lib/traject_config.rb [hunk 3]
```diff
@@ -1155,57 +1161,101 @@
...
+  MarcExtractor.cached("AVA").collect_matching_lines(record) do |field, _spec, _extractor|
     holding_b = nil
     is_alma = alma_code?(field['8'])
-    is_scsb = scsb_doc?(record['001'].value)
+    is_scsb = scsb_doc?(record_001)
+
     field.subfields.each do |s_field|
-      # Alma::skip any 852 fields that do not have subfield 8 with a value that begins with 22
+      # Alma::skip any AVA fields that do not have subfield 8 with a value that begins with 22
       if s_field.code == 'b'
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1178]: The comments should be 852 not AVA.

File: marc_to_solr/lib/traject_config.rb [hunk 4]
```diff
@@ -1305,43 +1355,32 @@
 # Position 852|k in the beginning of the call_number_display
 # The call_number_display is used in the catalog record page.
 to_field 'call_number_display' do |record, accumulator|
-  MarcExtractor.cached('852hik').collect_matching_lines(record) do |field, _spec, _extractor|
-    accumulator << [field['k'], field['h'], [field['i']]].compact.join(" ")
+  MarcExtractor.cached('AVAd').collect_matching_lines(record) do |field, _spec, _extractor|
+    accumulator << field['d']
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1359]: Same here. Please reverse the changes to 852.

File: marc_to_solr/lib/traject_config.rb [hunk 5]
```diff
@@ -1305,43 +1355,32 @@
...
 end
 
 # Position 852|k at the end of the call_number_browse_s
 # The call_number_browse_s is used in the call number browse page in the catalog
 to_field 'call_number_browse_s' do |record, accumulator|
-  MarcExtractor.cached('852hik').collect_matching_lines(record) do |field, _spec, _extractor|
-    accumulator << [field['h'], field['i'], [field['k']]].compact.join(" ")
+  MarcExtractor.cached('AVAd').collect_matching_lines(record) do |field, _spec, _extractor|
+    accumulator << field['d']
   end
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1368]: Please reverse to 852.

File: marc_to_solr/lib/traject_config.rb [hunk 6]
```diff
@@ -1305,43 +1355,32 @@
...
 end
 
 # The call_number_locator_display is used in the 'Where to find it' feature in the record page,
 # when the location is firestone$stacks.
-to_field 'call_number_locator_display', extract_marc('852hi')
+to_field 'call_number_locator_display', extract_marc('AVAd')
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1374]: Please reverse to 852.

File: marc_to_solr/lib/traject_config.rb [hunk 7]
```diff
@@ -1305,43 +1355,32 @@
...
 
 to_field 'electronic_portfolio_s' do |record, accumulator|
   fields = []
-  dates = []
-  embargoes = []
-  MarcExtractor.cached('951knx').collect_matching_lines(record) do |field, _spec, _extractor|
+  MarcExtractor.cached('AVEmsu').collect_matching_lines(record) do |field, _spec, _extractor|
     fields << field
```
Comment (christinach) [inline marc_to_solr/lib/traject_config.rb:1378]: Same here. AVE is a tag that comes from the Alma api call. In the exported files the data enrichment has 951 field. Please reverse the change.

File: spec/fixtures/marc_to_solr/991583506421.mrx [hunk 8]
```diff
@@ -42,145 +41,120 @@
...
-    <subfield code="a">No. 1 (Feb. 1975)-no. 68</subfield>
-    <subfield code="8">22262098640006421</subfield>
+  <datafield ind1=" " ind2=" " tag="INST">
+    <subfield code="a">01PRI_INST</subfield>
   </datafield>
-  <datafield tag="866" ind1=" " ind2="0">
-    <subfield code="z">LACKS: no. 25-26,29,49-50, 56</subfield>
+  <datafield ind1=" " ind2=" " tag="AVA">
+    <subfield code="0">991583506421</subfield>
     <subfield code="8">22262098640006421</subfield>
```
Comment (christinach) [inline spec/fixtures/marc_to_solr/991583506421.mrx:144]: I'm assuming you created the mrx using the bibdata bibliographic endpoint. Please use the sets in alma to export the desired fixtures.

File: spec/fixtures/marc_to_solr/alma/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+* locations (https://bibdata.princeton.edu/bibliographic/994264203506421)
+* call_number_locator (https://bibdata.princeton.edu/bibliographic/9941598513506421)
+* access_links (https://bibdata.princeton.edu/bibliographic/9918573506421)
+* electronic_record (https://bibdata.princeton.edu/bibliographic/99122424622606421)
+* call_number_display (https://bibdata.princeton.edu/bibliographic/9957270023506421)
+* electronic_portfolio (https://bibdata.princeton.edu/bibliographic/99122306151806421)
+* contained_in (https://bibdata.princeton.edu/bibliographic/9939073273506421)
+* elf_codes (https://bibdata.princeton.edu/bibliographic/99121576653506421)
+* holdings (https://bibdata.princeton.edu/bibliographic/9946093213506421)
+
```
Comment (christinach) [inline spec/fixtures/marc_to_solr/alma/README.md:32]: The mms ids should be used to export the desired fixtures from alma back end.

**Review by christinach — CHANGES_REQUESTED** [review]
@jrgriffiniii I can show you how to export fixtures from alma. My understanding is that you used the bibliographic endpoint which lead to all the unnecessary changes in princeton_marc and traject_config. The fixtures for the indexing should be exported from the alma back end.

**Comment by hackartisan** [comment]
replaced by https://github.com/pulibrary/bibdata/pull/1491

**Pull request closed without merging**